internal/compliance/resource_processor/processor/handler.go

```
@@ -82,6 +83,8 @@ func Handle(ctx context.Context, batch *events.SQSEvent) (err error) {
 		} else if resource != nil {
 			// Resource updated - analyze with applicable policies (after grouping + deduping)
 			resources[string(resource.ID)] = resource
+		} else {
+			zap.L().Error("failed to parse msg as resource, policy, or resource lookup", zap.String("body", record.Body))
```

Previously in this case we just continued looping, but that can lead to a situation where all messages failed to parse but we still make a GET policies request from the `analysis-api` for no reason. We could return an error here if we're ok with aborting the whole request based on one bad input, or we could have a check in place below that at least one resource is present before continuing.